src/meltano/core/project_init_service.py

```diff
@@ -43,9 +44,17 @@ def init(self, activate: bool = True, add_discovery: bool = False) -> Project:
         """
         try:
             os.mkdir(self.project_name)
-        except Exception as exp:  # noqa: F841
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise ProjectInitServiceError(
+                    f"Directory {self.project_name} already exists."
+                )
+            elif e.errno in {errno.EPERM, errno.EACCES}:
```

Comment:
Why use `e.errno` instead of checking for `FileExistsError`/`PermissionError`?

Reply:
I was under the impression that all of the methods in the `os` module only raise 'OSError`. I'll look into checking the other errors.